generators/client/templates/react/src/main/webapp/app/modules/administration/metrics/metrics-modal.tsx.ejs

```diff
@@ -1,57 +1,161 @@
+  showModal: boolean;
+  handleClose: Function;
+  threadDump: any;
+}
+
+export interface IMetricsModalState {
+  badgeFilter: string;
+  searchFilter: string;
+}
+
+export class MetricsModal extends React.Component<IMetricsModalProps, IMetricsModalState> {
+  state: IMetricsModalState = {
+    badgeFilter: '',
+    searchFilter: ''
+  };
+
+  computeFilteredList = () => {
+    const { badgeFilter, searchFilter } = this.state;
+    let filteredList = this.props.threadDump.threads;
+    if (badgeFilter !== '') {
+      filteredList = filteredList.filter(t => t.threadState === badgeFilter);
+    }
+    if (searchFilter !== '') {
+      filteredList = filteredList.filter(t => t.lockName != null &&
+                                              t.lockName.toLowerCase().includes(searchFilter.toLowerCase()));
+    }
+    return filteredList;
+  };
+
+  computeCounters = () => {
+    let threadDumpAll = 0;
+    let threadDumpRunnable = 0;
+    let threadDumpWaiting = 0;
+    let threadDumpTimedWaiting = 0;
+    let threadDumpBlocked = 0;
+
+    this.props.threadDump.threads.forEach(t => {
+      switch (t.threadState) {
+        case 'RUNNABLE':
+          threadDumpRunnable++;
+          break;
+        case 'WAITING':
+          threadDumpWaiting++;
+          break;
+        case 'TIMED_WAITING':
+          threadDumpTimedWaiting++;
+          break;
+        case 'BLOCKED':
+          threadDumpBlocked++;
+          break;
+        default:
+          break;
+      }
+    });
+
+    threadDumpAll = threadDumpRunnable + threadDumpWaiting + threadDumpTimedWaiting + threadDumpBlocked;
+    return { threadDumpAll, threadDumpRunnable, threadDumpWaiting, threadDumpTimedWaiting, threadDumpBlocked };
+  };
+
+  getBadgeClass = threadState => {
+    if (threadState === 'RUNNABLE') {
+      return 'badge-success';
+    } else if (threadState === 'WAITING') {
+      return 'badge-info';
+    } else if (threadState === 'TIMED_WAITING') {
+      return 'badge-warning';
+    } else if (threadState === 'BLOCKED') {
+      return 'badge-danger';
+    }
+  };
+
+  updateBadgeFilter = badge => () => {
+    this.setState({ badgeFilter: badge });
+  };
+
+  updateSearchFilter = () => event => {
+    this.setState({ searchFilter: event.target.value });
+  };
+
+  render() {
+    const { showModal, handleClose, threadDump } = this.props;
+    let counters = null;
+    let filteredList = null;
+    if (threadDump && threadDump.threads) {
+      counters = this.computeCounters();
+      filteredList = this.computeFilteredList();
+    }
+
+    return <Modal
+      isOpen={showModal} modalTransition={{ timeout: 20 }} backdropTransition={{ timeout: 10 }}
+      toggle={handleClose} className="modal-lg">
+      <ModalHeader toggle={handleClose}>Threads dump</ModalHeader>
+      <ModalBody>
+        <Badge color="primary" onClick={this.updateBadgeFilter('')}>All&nbsp;
+          <span className="badge badge-pill badge-default">{counters ? counters.threadDumpAll : null}</span>
```

Comment:
you can avoid the ternary by initializing `counters = {}` so this can be just `{ counters.threadDumpAll || 0 }`